### ISSUE TITLE
Conditional Compilation

### ISSUE THREAD

**Issue by HzanRsxa2959** [sannybuilder/dev#144]
Add a way to perform conditional compilation: https://docs.microsoft.com/en-us/cpp/preprocessor/hash-if-hash-elif-hash-else-and-hash-endif-directives-c-cpp?view=msvc-160

I don't know how common of a case this is to warrant such a feature, but I realized the need for it here: https://forum.mixmods.com.br/f16-utilidades/t5341-cleo-property-template

**Comment by x87** [comment]
I think this idea was expressed in https://github.com/sannybuilder/dev/issues/38

**Comment by HzanRsxa2959** [comment]
Yes, you understood correctly here: https://github.com/sannybuilder/dev/issues/38#issuecomment-664421979.
Although the original request was to make compilation depend on edit modes, I would prefer it if it could instead depend on constants, like here in my code:
`/* interior teleport >`
`...`
`// < */`
The user could just modify a constant instead of browsing the code and commenting/uncommenting the block ( that is if they even see the block before compiling).

**Comment by x87** [comment]
Yes, the user should be able to define its own variables to control pre-processor behavior, as well as have access to built-in variables provided by the current environment (e.g. the currently selected edit mode).
